### PR TITLE
main(): improve error handling and clean up code

### PR DESCRIPTION
**Pull request by M374LX**
Improve command-line error handling and clean up code in `main()`.

Adjustments have been done also in code conventions and capitalization of command-line output.

### CODE REVIEW

File: src/core/main.cpp [hunk 1]
```diff
@@ -120,36 +120,36 @@ void printHelp()
 {
 	printf( "LMMS %s\n"
 		"Copyright (c) 2004-%s LMMS developers.\n\n"
-		"usage: lmms [ -r <project file> ] [ options ]\n"
+		"Usage: lmms [ -r <project file> ] [ options ]\n"
```
Comment (tresf) [inline src/core/main.cpp]: `usage:` tends to be a lowercase word in other unix programs, as does the first letter of the sentence from what I'm observing.

Comment (M374LX) [inline src/core/main.cpp]: In my Debian 8 system, passing `--help` to commands like `rm`, `cp`, `ls`, and `mkdir` outputs the word `Usage` in uppercase.

Comment (Wallacoloo) [inline src/core/main.cpp]: @M374LX it tends to be uppercase for me as well (Ubuntu) - tested with `rm`, `rsync`, `firefox`, `more`, `subl` (Sublime Text), `gedit`. `tput` is the only one I found that uses a lowercase convention.

Comment (tresf) [inline src/core/main.cpp]: I probably should have checked Linux, I was on a BSD flavor at the time of trying.  It looks better your way anyway. :+1:

**Comment by Wallacoloo** [comment]
Code looks good. I tested it somewhat thoroughly & it doesn't appear to introduce any new bugs. Therefore merging.

By the way, I came across a really nice looking argument parsing library: [docopt](https://github.com/docopt/docopt.cpp), though it requires gcc >= 4.9. We should consider making use of it at some point in the future though :-)

**Comment by Wallacoloo** [comment]
Sorry, I probably should have let you squash those commits first. Oh well!

**Pull request merged**